rudder-web/src/main/scala/com/normation/rudder/web/snippet/administration/ClearCache.scala

```diff
@@ -101,9 +102,55 @@ class ClearCache extends DispatchSnippet with Loggable {
           }
           logger.debug("Deleting node configurations on user clear cache request")
           asyncDeploymentAgent ! AutomaticStartDeployment(modId, CurrentUser.getActor)
-          Full(set)
-      }
+        }
+        Full(set)
+    }
   }
+
+  /*
+   * This method clear all caches, which are:
+   * - node configurations (force full regen)
+   * - cachedAgentRunRepository
+   * - recentChangesService
+   * - reportingServiceImpl
+   * - nodeInfoServiceImpl
+   */
+  def action =  {
+    S.clearCurrentNotices
+    val modId = ModificationId(uuidGen.newUuid)
+
+    //clear agentRun cache
+    clearableCache.foreach { _.clearCache }
+
+    //clear node configuration cache
+    (for {
+      set <- clearNodeConfigurationCache(storeEvent = false)
+      _   <- eventLogRepository.saveEventLog(
+                modId
+              , ClearCacheEventLog(
+                  EventLogDetails(
+                      modificationId = Some(modId)
+                    , principal = CurrentUser.getActor
+                    , details = EventLog.emptyDetails
+                    , reason = Some("Clearing cache for: node configuration, recent changes, compliance and node info at user request")
+                  )
+                )
+             )
+    } yield {
+      set
+    }) match {
+      case eb:EmptyBox =>
+        val e = eb ?~! "Error when logging the cache event"
```

Comment:
The error may not be wile logging the cache event but also when clearNodeConfigurationCache fails